### PR TITLE
Incorporate pl-r-helpers into main PrairieLearn repository

### PR DESCRIPTION
**Pull request by barbehenna**
By incorporating the `PrairieLearn/pl-r-helpers` repository into `PrairieLearn` we avoid the need to request manual rebuilds of the external autograding image (`prairielearn/grader-r`) whenever `PrairieLearn/pl-r-helpers` is updated. Furthermore, this makes the structure of `graders/r` more similar to the structure of `graders/python`.

I copied the git history from `PrairieLearn/pl-r-helpers` with `git format-patch` and `git am --directory=graders/r/plr`. The final commit is original to this PR.

### CODE REVIEW

**Comment by nwalters512** [comment]
@daviddalpiaz you'll need to sign the CLA before we can accept this - see the comment above!

**Comment by daviddalpiaz** [comment]
I have reviewed and hereby sign the CLA.

@nwalters512

**Comment by eddelbuettel** [comment]
recheck

**Comment by daviddalpiaz** [comment]
I have reviewed and hereby sign the CLA

**Comment by daviddalpiaz** [comment]
(Re-posting in case tagging Nathan ruined the formatting and broke an automated process)

File: graders/r/plr/man/get_question_details.Rd [hunk 1]
```diff
@@ -0,0 +1,24 @@
+% Generated by roxygen2: do not edit by hand
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply graders/r/plr/man/get_question_details.Rd:1]: Yes. That is very standard for R.

In fact, R itself requires (and only sees) `man/*.Rd`.  That another tool helped make them is immaterial.

[the remark this answers is not in the document]
Comment (daviddalpiaz) [reply graders/r/plr/man/get_question_details.Rd:1]: Yes.

(OK, not *strictly* necessary, but if they were removed, and a user installed the package for local testing purposes, the functions would lack documentation. @eddelbuettel can correct me if I'm wrong.)

File: graders/r/Dockerfile [hunk 2]
```diff
@@ -84,3 +84,7 @@ RUN useradd ag \
 
 # Copy autograder interface script into position
 COPY r_autograder /r_autograder
+
+# Install plr (helper package for grading R code in PrairieLearn)
+COPY plr /r_autograder/plr
+RUN R CMD INSTALL r_autograder/plr
```
Comment (eddelbuettel) [inline graders/r/Dockerfile:90]: We could make this line 75 instead and write `RUN install.r r_autograder/plr` to use the corresponding script from littler but that is really a marginal change.

File: graders/r/plr/ChangeLog [hunk 3]
```diff
@@ -0,0 +1,50 @@
+2021-08-31  Dirk Eddelbuettel  <edd@debian.org>
```
Comment (eddelbuettel) [inline graders/r/plr/ChangeLog:1]: We could continue to maintain a ChangeLog, or decide we're done with it (and live by git commit messages) and delete this one.

File: graders/r/plr/DESCRIPTION [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Package: plr
+Title: Utility Functions for 'PrairieLearn' and R
+Version: 0.1.0
+Date: 2021-08-17
+Author: Dirk Eddelbuettel and Alton Barbehenn
```
Comment (eddelbuettel) [inline graders/r/plr/DESCRIPTION:5]: add:   and David Dalpiaz

File: graders/r/plr/DESCRIPTION [hunk 5]
```diff
@@ -0,0 +1,18 @@
+Package: plr
+Title: Utility Functions for 'PrairieLearn' and R
+Version: 0.1.0
+Date: 2021-08-17
```
Comment (eddelbuettel) [inline graders/r/plr/DESCRIPTION:4]: We could update the version, but it no longer matters much as it is in the same repo.  A Date: is optional for R so we could remove it.  (I tend to keep them, just as a reminder to myself...)

**Review by eddelbuettel — CHANGES_REQUESTED** [review]
Looks very good.  I have a couple of micro nits to pick which each individually amount to little, but in aggregate may warrant another round (esp adding David to Authors:).

All the code functionality is unchanged and trusted, no comments or issues there.

File: graders/r/plr/DESCRIPTION (hunk 4 above)
Comment (barbehenna) [reply graders/r/plr/DESCRIPTION:5]: Definitely an oversight!

File: graders/r/plr/DESCRIPTION (hunk 5 above)
Comment (barbehenna) [reply graders/r/plr/DESCRIPTION]: Updated to date of version bump

File: graders/r/Dockerfile (hunk 2 above)
Comment (barbehenna) [reply graders/r/Dockerfile]: Done.

File: graders/r/plr/ChangeLog (hunk 3 above)
Comment (barbehenna) [reply graders/r/plr/ChangeLog:11]: Updated.

**Review by eddelbuettel — APPROVED** [review]
LGTM!

**Comment by eddelbuettel** [comment]
Thanks for updating once more and syncing with master, @barbehenna !

**Comment by barbehenna** [comment]
I added a small commit to make the grading environment cleaner and to remove the following message from line 44 of the `run.sh` file.
```
info: container> cp: -r not specified; omitting directory '/r_autograder/plr' 
```

**Pull request merged**